src/injected/visualization/svg-shape-factory.ts

```diff
@@ -97,11 +98,42 @@ export class SVGShapeFactory {
 
         if (tabOrder != null) {
             text.innerHTML = tabOrder.toString();
+        } else {
+            text.innerHTML = 'X';
         }
 
         return text;
     }
 
+    public createFailureLabel(center: Point, failureBoxConfig: FailureBoxConfig): Element {
+        const myDocument = this.drawerUtils.getDocumentElement();
+
+        const box = myDocument.createElementNS(SVGNamespaceUrl, 'rect');
+        box.classList.add('insights-highlight-text');
+        box.classList.add('failure-label');
+        box.setAttributeNS(null, 'x', (center.x + 10).toString());
```

Comment:
Curious how you came up with the calculations for the position?

Reply:
Yeah, the circles are 10px in radius, so to get to the upper right corner we need to do center.x + 10 and center.y - 10. From there, these numbers have to be adjusted to account for the size of the red box, and then to center the '!' in the box.